### PR TITLE
Support https

### PR DESCRIPTION
**Pull request by carvinlo**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [ ] My code follows the code style of this project
- [ ] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

[Place a meaningful description here]

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

File: src/tooling/piral-cli/src/apps/debug-pilet.ts [hunk 1]
```diff
@@ -204,6 +204,7 @@ export async function debugPilet(baseDir = process.cwd(), options: DebugPiletOpt
   }
 
   const injectorConfig = {
+    ssl: krasConfig.ssl,
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/apps/debug-pilet.ts:207]: I don't think thats necessary -> see below.

File: src/tooling/piral-cli/src/injectors/pilet.ts [hunk 2]
```diff
@@ -20,20 +20,22 @@ export interface PiletInjectorConfig extends KrasInjectorConfig {
 export default class PiletInjector implements KrasInjector {
   public config: PiletInjectorConfig;
   private port: number;
+  private https: boolean;
 
   constructor(options: PiletInjectorConfig, config: KrasConfiguration, core: EventEmitter) {
     this.config = options;
     this.port = config.port;
+    this.https = !!config.ssl;
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/injectors/pilet.ts:28]: Yes this one adds it already, so no need to have it in the injector config, too.

File: src/tooling/piral-cli/src/injectors/pilet.ts [hunk 3]
```diff
@@ -20,20 +20,22 @@ export interface PiletInjectorConfig extends KrasInjectorConfig {
 export default class PiletInjector implements KrasInjector {
   public config: PiletInjectorConfig;
   private port: number;
+  private https: boolean;
```
Comment (FlorianRappl) [inline src/tooling/piral-cli/src/injectors/pilet.ts:23]: Alternatively, call this `protocol` and have it set to either `http` or `https` -> then we can spare the conditional below and evaluate it only once.

File: src/tooling/piral-cli/src/apps/debug-pilet.ts (hunk 1 above)
Comment (carvinlo) [reply src/tooling/piral-cli/src/apps/debug-pilet.ts:207]: Thanks for pointing right

File: src/tooling/piral-cli/src/injectors/pilet.ts (hunk 2 above)
Comment (carvinlo) [reply src/tooling/piral-cli/src/injectors/pilet.ts:28]: Thanks for pointing right

File: src/tooling/piral-cli/src/injectors/pilet.ts (hunk 3 above)
Comment (carvinlo) [reply src/tooling/piral-cli/src/injectors/pilet.ts:23]: ok

**Comment by carvinlo** [comment]
Clean up the commit, pull a new [PR](https://github.com/smapiot/piral/pull/274).

**Pull request closed without merging**